cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -725,8 +728,10 @@ services::Status TrainBatchTaskBase<algorithmFPType, BinIndexType, DataHelper, H
     DAAL_CHECK_MALLOC(_aSample.get() && _helper.reset(_nSamples) && _helper.resetWeights(_nSamples) && _aFeatureBuf.get() && _aFeatureIndexBuf.get()
                       && _aFeatureIdx.get());
 
+#if defined(DAAL_INTEL_CPP_COMPILER)
     PRAGMA_IVDEP
     PRAGMA_VECTOR_ALWAYS
+#endif
```

Comment:
Please remove those pragmas, as the loop contains memory allocation functions calls and is not vectorized anyway.
The optimization report confirms that vectorization does not happen:
```
LOOP BEGIN at ./cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i (730, 5)
LOOP END
```